packages/web-components/fast-components/src/tabs/tabs.ts

```diff
@@ -0,0 +1,324 @@
+    public handleEndContentChange(): void {
+        this.end.assignedNodes().length > 0
+            ? this.endContainer.classList.add("end")
+            : this.endContainer.classList.remove("end");
+    }
+
+    public change = (): void => {
+        this.$emit("change", this.activeTab);
+    };
+
+    @attr({ mode: "boolean" })
+    public activeIndicator = true;
+
+    @observable
+    public activeIndicatorOffset: number = 0;
+    public activeIndicatorRef: HTMLElement;
+    public activeTabIndex: number = 0;
+    public activeTab: HTMLElement;
+
+    private prevActiveTabIndex: number = 0;
+    private ticking: boolean = false;
+    private tabIDs: string[];
+    private tabPanelIDs: string[];
+    private connected: boolean = false;
+
+    private getActiveIndex(): number {
+        if (this.activeId !== undefined) {
+            return this.tabIDs.indexOf(this.activeId) === -1
+                ? 0
+                : this.tabIDs.indexOf(this.activeId);
+        } else {
+            return 0;
+        }
+    }
+
+    private getTabs = (): void => {
+        this.tabIDs = this.getTabIds();
+        this.tabPanelIDs = this.getTabPanelIds();
+        this.activeTabIndex = this.getActiveIndex();
+        this.tabs.forEach((tab: HTMLElement, index: number) => {
+            if (tab.slot === "tab") {
+                tab.setAttribute(
+                    "id",
+                    this.tabIDs[index] === undefined || this.tabIDs[index] === null
+                        ? `tab-${index + 1}`
+                        : this.tabIDs[index]
+                );
+                tab.setAttribute(
+                    "aria-selected",
+                    this.activeTabIndex === index ? "true" : "false"
+                );
+                tab.setAttribute(
+                    "aria-controls",
+                    this.tabPanelIDs[index] === undefined ||
+                        this.tabPanelIDs[index] === null
+                        ? `panel-${index + 1}`
+                        : this.tabPanelIDs[index]
+                );
+                tab.setAttribute(
+                    "style",
+                    this.isHorizontal()
+                        ? `grid-column: ${index + 1};`
+                        : `grid-row: ${index + 1};`
+                );
+                tab.addEventListener("click", this.handleTabClick);
+                tab.addEventListener("keydown", this.handleTabKeyDown);
+                tab.setAttribute("tabindex", this.activeTabIndex === index ? "0" : "-1");
+                if (this.activeTabIndex === index) {
+                    this.activeTab = tab;
+                }
+                !this.isHorizontal()
+                    ? tab.classList.add("vertical")
+                    : tab.classList.remove("vertical");
+            }
+        });
+    };
+
+    private getTabPanels = (): void => {
+        this.tabIDs = this.getTabIds();
+        this.tabPanelIDs = this.getTabPanelIds();
```

Comment:
Can you rename `getTabPanels` and `getTabs` to something more descriptive of what they're doing? They don't actually return anything and seem to have numerous side-effects